src/modelAnalysis/FBA/optimizeCbModel.m

```diff
@@ -194,40 +199,85 @@
 %use global solver parameter for printLevel
 [printLevel,primalOnlyFlag] = getCobraSolverParams('LP',{'printLevel','primalOnly'});
 
+% Rest of the LP problem
 [nMets,nRxns] = size(model.S);
-
-% add csense
-%Doing this makes csense a double array.  Totally smart design move.
-%LPproblem.csense = [];
-if ~isfield(model,'csense')
-    % If csense is not declared in the model, assume that all
-    % constraints are equalities.
-    if printLevel>1
-        fprintf('%s\n','LP problem has no defined csense. We assume that all constraints are equalities.')
-    end
-    LPproblem.csense(1:nMets,1) = 'E';
-else % if csense is in the model, move it to the lp problem structure
-    if length(model.csense)~=nMets,
-        warning('Length of csense is invalid! Defaulting to equality constraints.')
-        LPproblem.csense(1:nMets,1) = 'E';
+if isfield(model,'C')
+    if isfield(model,'d')
+        [nIneq,nltC]=size(model.C);
+        [mltd,nltd]=size(model.d);
+        if nltC~=nRxns
+            error('For the constraints C*v <= d, the number of columns of S and C are inconsisent')
+        end
+        if nIneq~=mltd
+            error('For the constraints C*v <= d, the number of rows of C and d are inconsisent')
+        end
+        if nltd~=1
+            error('For the constraints C*v <= d, d must have only one column')
+        end
+        LPproblem.A = [model.S;model.C];
+        % Fill in the RHS vector if not provided
+        if ~isfield(model,'b')
+            warning('LP problem has no defined b in S*v=b. b should be defined, for now we assume b=0')

```

Comment:
addRatioReaction should be moved to this coupling folder then we ultimately need a tutorial explaining how to use them. Each should make C*v <=> d constraints that optimizeCbModel recognises.